src/main/java/hybrid/it/internship/library/service/impl/RentServiceImpl.java

```diff
@@ -21,4 +42,89 @@ public boolean existsByBookId(Long bookId) {
     public boolean existsByUserId(Long userId) {
         return rentRepository.existsByUserId(userId);
     }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<RentDTO> getOverdueRents() {
+
+        List<RentDTO> rentsDTOs = rentMapper.toDTO(rentRepository.findAll());
+
+        rentsDTOs.removeIf(rent -> !rent.getRentDate().plusDays(rentPeriodInDays).isBefore(LocalDateTime.now()));
+
+        return rentsDTOs;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<RentedBookDTO> getMostRented() {
```

Comment:
Definitely go for an SQL query. It is much more performant and will simplify the logic by removing the need to sort and limit the elements.